### PR TITLE
Filter metadata

### PR DESCRIPTION
**Pull request by bklare**
A simple transform for filtering based on metadata.

### CODE REVIEW

File: openbr/plugins/metadata/filtermetadata.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#include <openbr/plugins/openbr_internal.h>
+
+namespace br
+{
+
+/*!
+ * \ingroup transforms
+ * \brief Filters templates such that the only remaining template wills have metadata values witihin the
+ *          the specified ranges. 
+ * @param[in] key1,key2 The meta-data key(s) to filter on
```
Comment (jklontz) [inline openbr/plugins/metadata/filtermetadata.cpp]: @JordanCheney does the doc engine care about `\param` vs. `@param`?

Comment (JordanCheney) [inline openbr/plugins/metadata/filtermetadata.cpp]: Right now the doc engine parses only ```\br_property``` which I am using as a replacement for param.

**Comment by jklontz** [comment]
@bklare LGTM pending doc fix

**Comment by bklare** [comment]
Thanks for the clarification. Just updated.

**Comment by JordanCheney** [comment]
@bklare one more change ```\br_property```expects input as ```\br_property type name description``` so in your case ```\br_property QString key1 description```. Sorry for the confusion.

**Comment by bklare** [comment]
@JordanCheney Thanks for the clarification.

@jklontz Should be GTG.

**Pull request merged**